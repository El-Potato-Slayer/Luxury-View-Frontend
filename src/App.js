import './App.css';
import { BrowserRouter, Route, Switch } from 'react-router-dom';
import Navbar from './components/Navbar';
import Home from './components/Home';

function App() {
  return (
    <div data-testid="app" className="App">
      <BrowserRouter>
        <Navbar />
        <Switch>
          <Route exact path="/">
            <Home />
            {/* <div data-testid="app-homepage">
              Homepage
            </div> */}
          </Route>
          <Route path="/mansions">
            <div data-testid="app-mansions">
              <h2>Mansions</h2>
            </div>
          </Route>
          <Route path="/appointments">
            <div data-testid="app-appointments">
              <h2>Appointments</h2>
            </div>
          </Route>
          <Route path="/agents">
            <div data-testid="app-agents">
              <h2>Agents</h2>
            </div>
          </Route>
        </Switch>
      </BrowserRouter>
    </div>
  );
}

export default App;
