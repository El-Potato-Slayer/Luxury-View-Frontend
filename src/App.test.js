import {
  cleanup, fireEvent, render, screen,
} from '@testing-library/react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import App from './App';
import '@testing-library/jest-dom';

afterEach(cleanup);

test('renders learn react link', () => {
  // render(<App />);
  // const linkElement = screen.getByText(/learn react/i);
  // expect(linkElement).toBeInTheDocument();
  console.log(123);
});

it('Renders App component without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
});

test('Renders the homepage', () => {
  render(
    <App />,
  );
  fireEvent.click(screen.getByTestId('navbar-homepage'));
  expect(screen.getByTestId('homepage')).toHaveTextContent('Luxury View');
});

test('Renders mansions component', () => {
  render(
    <App />,
  );
  fireEvent.click(screen.getByTestId('navbar-mansions'));
  expect(screen.getByTestId('app-mansions')).toHaveTextContent('Mansions');
});

test('Renders appointments component', () => {
  render(
    <App />,
  );
  fireEvent.click(screen.getByTestId('navbar-appointments'));
  expect(screen.getByTestId('app-appointments')).toHaveTextContent('Appointments');
});

test('Renders agents component', () => {
  render(
    <App />,
  );
  fireEvent.click(screen.getByTestId('navbar-agents'));
  expect(screen.getByTestId('app-agents')).toHaveTextContent('Agents');
});

it('matches snapshot', () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});
