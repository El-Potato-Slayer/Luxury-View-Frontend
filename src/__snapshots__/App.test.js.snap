// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="App"
  data-testid="app"
>
  <div
    className="navbar"
    data-testid="navbar"
  >
    <a
      data-testid="navbar-homepage"
      href="/"
      onClick={[Function]}
    >
      Home
    </a>
    <a
      data-testid="navbar-mansions"
      href="/mansions"
      onClick={[Function]}
    >
      Mansions
    </a>
    <a
      data-testid="navbar-appointments"
      href="/appointments"
      onClick={[Function]}
    >
      Appointments
    </a>
    <a
      data-testid="navbar-agents"
      href="/agents"
      onClick={[Function]}
    >
      Agents
    </a>
  </div>
  <div
    data-testid="app-agents"
  >
    <h2>
      Agents
    </h2>
  </div>
</div>
`;
